### PR TITLE
Rename overflowing_{add,sub,mul} intrinsics to wrapping_{add,sub,mul}.

### PR DESCRIPTION
**Pull request by eddyb**
These confused @Gankra, and then, also me, especially since `overflowing_*` *methods* also exist, but they map to `*_with_overflow` intrinsics!

r? @oli-obk / @nikomatsakis cc @Mark-Simulacrum (on the rustbuild workaround)

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Bootstrap changes look fine.

File: src/librustc_typeck/check/intrinsic.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ pub fn intrisic_operation_unsafety(intrinsic: &str) -> hir::Unsafety {
     match intrinsic {
         "size_of" | "min_align_of" | "needs_drop" |
         "add_with_overflow" | "sub_with_overflow" | "mul_with_overflow" |
```
Comment (Gankra) [inline src/librustc_typeck/check/intrinsic.rs]: should these be changed to match the others, or is their different signature sufficient to motivate a different naming convention? (this naming convention is not reflected in the std names)

File: src/librustc_mir/transform/qualify_min_const_fn.rs [hunk 2]
```diff
@@ -379,9 +379,9 @@ fn is_intrinsic_whitelisted(tcx: TyCtxt<'tcx>, def_id: DefId) -> bool {
         | "add_with_overflow" // ~> .overflowing_add
```
Comment (Centril) [inline src/librustc_mir/transform/qualify_min_const_fn.rs]: While we're at it, maybe rename these also?

File: src/librustc_typeck/check/intrinsic.rs (hunk 1 above)
Comment (eddyb) [reply src/librustc_typeck/check/intrinsic.rs]: These match the LLVM intrinsics. We could rename them later but they're not *outright wrong* like `overflowing_*` intrinsics were.

**Comment by rkruppe** [comment]
r=me on the code changes, don't have an opinion on the rustbuild stuff though.

File: src/bootstrap/bin/rustc.rs [hunk 3]
```diff
@@ -102,8 +102,13 @@ fn main() {
...
     // being broken on the beta compiler with bootstrap passed, so this is a temporary workaround
     // (we've just snapped, so there are no cfg(bootstrap) related annotations in core).
-    if stage == "0" && crate_name != Some("core") {
-        cmd.arg("--cfg").arg("bootstrap");
+    if stage == "0" {
+        if crate_name != Some("core") {
+            cmd.arg("--cfg").arg("bootstrap");
+        } else {
+            // NOTE(eddyb) see FIXME above, except now we need annotations again in core.
+            cmd.arg("--cfg").arg("boostrap_stdarch_ignore_this");
```
Comment (cuviper) [inline src/bootstrap/bin/rustc.rs]: _boo_ strap? :ghost:

**Pull request merged**